### PR TITLE
Fix `BuildSrcApiChangePerformanceTest`

### PR DESCRIPTION
**Pull request by wolfs**
Fixes the above performance test.

### CODE REVIEW

**Review by donat — APPROVED** [review]
lgtm

File: subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractCrossBuildPerformanceTestRunner.groovy [hunk 1]
```diff
@@ -138,6 +138,10 @@ abstract class AbstractCrossBuildPerformanceTestRunner<R extends CrossBuildPerfo
 
         try {
             runAllSpecifications(results)
+        } catch (Exception e) {
+            // Print the exception here, so it is reported even when the reporting fails
+            e.printStackTrace()
```
Comment (donat) [inline subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AbstractCrossBuildPerformanceTestRunner.groovy:143]: 👍 Thanks for fixing this.

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcApiChangePerformanceTest.groovy [hunk 2]
```diff
@@ -77,16 +83,7 @@ class BuildSrcApiChangePerformanceTest extends AbstractCrossVersionPerformanceTe
 
     private static class CreateChangingClassMutator implements BuildMutator {
 
-        private final InvocationSettings settings
```
Comment (donat) [inline subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcApiChangePerformanceTest.groovy:80]: This doesn't look right, but I get the reasons behind it. We should eventually fix this behavior. How about a TODO comment explaining that the code belongs to `beforeScenario`, it just doen't work atm.

**Review by donat — APPROVED** [review]

File: subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcApiChangePerformanceTest.groovy (hunk 2 above)
Comment (donat) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcApiChangePerformanceTest.groovy:80]: (I meant to select the entire `beforeScenaro` -> constructor conversion code above)

Comment (wolfs) [reply subprojects/performance/src/performanceTest/groovy/org/gradle/performance/regression/inception/BuildSrcApiChangePerformanceTest.groovy:80]: I added a comment. Please take another look.

**Review by donat — APPROVED** [review]
LGTM

**Pull request merged**